Lib/argparse.py

```diff
@@ -1290,21 +1292,36 @@ def __repr__(self):
 class Namespace(_AttributeHolder):
     """Simple object for storing attributes.
 
+    Default values are stored in a dict named `__defaults__` so they won't
+    override the given values.
+
     Implements equality by attribute names and values, and provides a simple
     string representation.
     """
 
     def __init__(self, **kwargs):
         for name in kwargs:
             setattr(self, name, kwargs[name])
+        self.__defaults__ = {}
+
+    def _get_kwargs(self):
+        kwargs = self.__defaults__ | self.__dict__
+        kwargs.pop('__defaults__', None)
+        return list(kwargs.items())
```

Comment:
Why do you convert it into a list if the caller (`__eq__`) turns it into a dict again?

Reply:
The `__repr__` implementation of the parent class `_AttributeHolder` also calls this. But this change can be considered.

Reply:
Ah right, I should have checked that. Better to just leave it as is then.